source/common/common/key_value_store_base.cc

```diff
@@ -1,5 +1,9 @@
 #include "source/common/common/key_value_store_base.h"
 
+#include <algorithm>
+#include <chrono>
+#include <optional>
```

Comment:
you'll want to include absl/types/optional.h, not the optional in the standard library